src/middleware/AdaptiveCardSubmitAction.ts

```diff
@@ -0,0 +1,24 @@
+import * as builder from "botbuilder";
+import { DialogMatches } from "../utils/DialogMatches";
+
+// Set the text field on message events to "adaptive card", if request is from an adaptive card
+export class AdaptiveCardSubmitAction implements builder.IMiddlewareMap {
+
+    public readonly receive = (event: builder.IEvent, next: Function): void => {
+        let currEvent = (event as any);
```

Comment:
I don't know if we need this cast does IEvent not have replyToId?

Reply:
the event has replyToId but doesn't have text and value property.

Reply:
Let's have this actually do one check just before this before the middleware tries to do anything - check that the type of IEvent is 'message' (if not just call next), and then if it is of type message, cast it to an IMessage which should have text and such.

Reply:
Done.